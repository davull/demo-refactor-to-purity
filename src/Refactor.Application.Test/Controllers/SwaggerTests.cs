using System.Net;
using FluentAssertions;
using Snapshooter.NUnit;

namespace Refactor.Application.Test.Controllers;

public class SwaggerTests : IntegrationTestBase
{
    [TestCase("/swagger")]
    [TestCase("/swagger/index.html")]
    [TestCase("/swagger/v1/swagger.json")]
    public async Task Should_Return_Ok(string path)
    {
        var response = await GetAsync(path);
        response.StatusCode.Should().Be(HttpStatusCode.OK);
    }

    [Test]
    public async Task Should_Match_Snapshot()
    {
        var content = await GetStringAsync("/swagger/v1/swagger.json");
        content.Should().MatchSnapshot();
    }
}