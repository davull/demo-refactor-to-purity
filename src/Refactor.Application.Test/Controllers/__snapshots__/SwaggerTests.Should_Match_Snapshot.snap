{
  "openapi": "3.0.1",
  "info": {
    "title": "Refactor.Application",
    "version": "1.0"
  },
  "paths": {
    "/Orders": {
      "get": {
        "tags": [
          "Orders"
        ],
        "parameters": [
          {
            "name": "startDate",
            "in": "query",
            "style": "form",
            "schema": {
              "type": "string",
              "format": "date-time"
            }
          },
          {
            "name": "endDate",
            "in": "query",
            "style": "form",
            "schema": {
              "type": "string",
              "format": "date-time"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": { }
}
